common/src/Utilities/PString.cpp

```diff
@@ -0,0 +1,95 @@
+#include "PrecompiledHeader.h"
+#include "PString.h"
+#include <codecvt>
+#include <wchar.h>
+
+const PString& PString::operator=(const PString& rhs)
+{
+	string = rhs.string;
+	return *this;
+}
+
+PString& PString::operator=(PString&& rhs)
+{
+	string = std::move(rhs.string);
+	return *this;
+}
+
+#ifdef _WIN32
+// Non default
+PString::PString(const std::wstring& str)
+{
+	char* temp = new char[count];
+	const wchar_t* wStr = str.c_str();
+	size_t len = (wcslen(wStr) + 1) * sizeof(wchar_t);
+	int err = wcstombs_s(&count, temp, len, wStr, len);
```

Comment:
`wcstombs` might not necessarily output the multibyte character encoding you want. If you want to ensure UTF-8, prefer using `WideCharToMultiByte` since it's a Windows only function anyway.